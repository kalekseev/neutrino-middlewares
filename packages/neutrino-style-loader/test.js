import { Neutrino } from 'neutrino';

// eslint-disable-next-line global-require, import/no-unresolved
const mw = () => require('.');
const options = { css: { modules: true }, style: { sourceMap: true } };

test('loads middleware', () => {
  mw();
});

test('uses middleware', () => {
  Neutrino().use(mw());
});

test('uses with options', () => {
  Neutrino().use(mw(), options);
});

test('instantiates', () => {
  const api = Neutrino();
  api.use(mw());
  const config = api.config.toConfig();
  expect(config).toMatchSnapshot();
});

test('instantiates with options', () => {
  const api = Neutrino();
  api.use(mw(), options);
  const config = api.config.toConfig();
  expect(config).toMatchSnapshot();
});
