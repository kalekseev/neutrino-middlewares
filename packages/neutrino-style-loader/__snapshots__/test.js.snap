// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instantiates 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "exclude": Array [
          /\\\\\\.module\\.s\\?css\\$/,
        ],
        "test": /\\\\\\.s\\?css\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/css-hot-loader/index.js",
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/extract-text-webpack-plugin/dist/loader.js",
            "options": Object {
              "omit": 1,
              "remove": true,
            },
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/style-loader/index.js",
            "options": Object {
              "hmr": true,
            },
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/css-loader/index.js",
            "options": Object {
              "importLoaders": 0,
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.module\\.s\\?css\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/css-hot-loader/index.js",
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/extract-text-webpack-plugin/dist/loader.js",
            "options": Object {
              "omit": 1,
              "remove": true,
            },
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/style-loader/index.js",
            "options": Object {
              "hmr": true,
            },
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/css-loader/index.js",
            "options": Object {
              "importLoaders": 0,
              "modules": true,
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [
    ExtractTextPlugin {
      "filename": "[name].css",
      "id": 1,
      "options": Object {
        "allChunks": true,
        "ignoreOrder": true,
      },
    },
    ExtractTextPlugin {
      "filename": "[name].css",
      "id": 2,
      "options": Object {
        "allChunks": true,
        "ignoreOrder": true,
      },
    },
  ],
}
`;

exports[`instantiates with options 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "exclude": Array [
          /\\\\\\.module\\.s\\?css\\$/,
        ],
        "test": /\\\\\\.s\\?css\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/css-hot-loader/index.js",
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/extract-text-webpack-plugin/dist/loader.js",
            "options": Object {
              "omit": 1,
              "remove": true,
            },
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/style-loader/index.js",
            "options": Object {
              "hmr": true,
              "sourceMap": true,
            },
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/css-loader/index.js",
            "options": Object {
              "importLoaders": 0,
              "modules": true,
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.module\\.s\\?css\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/css-hot-loader/index.js",
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/extract-text-webpack-plugin/dist/loader.js",
            "options": Object {
              "omit": 1,
              "remove": true,
            },
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/style-loader/index.js",
            "options": Object {
              "hmr": true,
              "sourceMap": true,
            },
          },
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/css-loader/index.js",
            "options": Object {
              "importLoaders": 0,
              "modules": true,
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [
    ExtractTextPlugin {
      "filename": "[name].css",
      "id": 3,
      "options": Object {
        "allChunks": true,
        "ignoreOrder": true,
      },
    },
    ExtractTextPlugin {
      "filename": "[name].css",
      "id": 4,
      "options": Object {
        "allChunks": true,
        "ignoreOrder": true,
      },
    },
  ],
}
`;
